storm-core/src/jvm/org/apache/storm/security/auth/SaslTransportPlugin.java

```diff
@@ -64,8 +64,14 @@ public void prepare(ThriftConnectionType type, Map storm_conf, Configuration log
     @Override
     public TServer getServer(TProcessor processor) throws IOException, TTransportException {
         int port = type.getPort(storm_conf);
+        Integer socketTimeout = type.getSocketTimeOut(storm_conf);
         TTransportFactory serverTransportFactory = getServerTransportFactory();
-        TServerSocket serverTransport = new TServerSocket(port);
+        TServerSocket serverTransport = null;
+        if (socketTimeout != null) {
```

Comment:
For NIMBUS thrift client it is not null but for DRPC there is a possibility of it being null
as per since the DRPC intialization here is null. Refering to https://github.com/apache/storm/pull/1837/files#diff-8a40b66d685fb8c7b58cfaba6272e14aR82 
and https://github.com/apache/storm/pull/1837/files#diff-8a40b66d685fb8c7b58cfaba6272e14aR82